zimui/src/stores/main.ts

```diff
@@ -1,52 +1,94 @@
 import { defineStore } from 'pinia'
-import axios from 'axios'
+import axios, { AxiosError }from 'axios'
 import Channel from '@/types/Channel'
 import Topic from '@/types/Topic'
 
 export type RootState = {
   channelData: Channel | null
   isLoading: boolean
   errorMessage: string
+  error: AxiosError<object> | null
 }
 export const useMainStore = defineStore('main', {
   state: () =>
     ({
       channelData: null,
       isLoading: false,
       errorMessage: '',
+      error: null,
     }) as RootState,
   getters: {},
   actions: {
     async fetchChannel() {
       this.isLoading = true
       this.errorMessage = ''
-      return axios.get('./channel.json').then(
-        (response) => {
-          this.isLoading = false
-          this.channelData = response.data as Channel
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./channel.json')
+        this.isLoading = false
+        this.channelData = response.data as Channel
+      } catch (error) {
+        if (error instanceof AxiosError) {
           this.isLoading = false
           this.channelData = null
-          this.errorMessage = 'Failed to load channel data'
-        },
-      )
+          this.handleAxiosError(error);
+          if (!axios.isAxiosError(error)) {
+            this.errorMessage = 'Failed to load channel data';
+          }
+          this.error = error // Set axios error to store
+        }
+      }
     },
     async fetchTopic(slug: string) {
       this.isLoading = true
       this.errorMessage = ''
-      return axios.get('./topics/' + slug + '.json').then(
-        (response) => {
-          this.isLoading = false
-          return response.data as Topic
-        },
-        (_) => {
+      try {
+        const response = await axios.get('./topics/' + slug + '.json')
+        this.isLoading = false
+        return response.data as Topic
+      } catch (error: unknown) {
+        if (error instanceof AxiosError) {
           this.isLoading = false
           this.channelData = null
-          this.errorMessage = 'Failed to load node ' + slug + ' data'
+          this.handleAxiosError(error);
+          if (!axios.isAxiosError(error)) {
+            this.errorMessage = 'Failed to load node ' + slug + ' data';
+          }
+          this.error = error // Set axios error to store
           return null
-        },
-      )
+        }
+      }
+    },
+    handleAxiosError(error: AxiosError<object>) {
+      this.isLoading = false;
+      this.channelData = null;
+      if (axios.isAxiosError(error)) {
+        if (error.response) {
+          const status = error.response.status;
+          switch (status) {
+            case 400:
+              this.errorMessage = 'Bad Request: The server could not understand the request due to invalid syntax.';
+              break;
+            case 401:
```

Comment:
There is not authentification in our case, this is useless

Reply:
Why did you marked this as resolved? I agreed to keep `500` errors, but `401` are still useless